src/codegen.cpp

```diff
@@ -4721,7 +4721,7 @@ static void emit_upsilonnode(jl_codectx_t &ctx, ssize_t phic, jl_value_t *val)
         }
         else if (vi.value.V && !vi.value.constant && vi.value.typ != jl_bottom_type) {
             assert(vi.value.ispointer());
-            Type *T = cast<AllocaInst>(vi.value.V)->getAllocatedType();
+            Type *T = cast<AllocaInst>(vi.value.V->stripPointerCasts())->getAllocatedType();
             if (CountTrackedPointers(T).count) {
```

Comment:
This seems now just to be i8, which would not be good

Reply:
Ok, so we should use `julia_type_to_llvm(vi.value.typ)` then?

Reply:
I am not entirely sure, since it seems like we might be losing the root here anyways. There was a thought that we might use the alloca type in the future to ensure we got all of the roots (even in the presence of stack-spills), but it was mostly never implemented.

Reply:
We might need to force this to become a boxroot

Reply:
It would be similarly good to know how the other branch of this change is reached, since typically we may lose gc-roots if they get shoved onto the stack, so we should not do that

Reply:
Let me change this to `julia_type_to_llvm(vi.value.typ)` for now, which will preserve current behavior and then we can separately track your concerns.

Reply:
Or I guess late-gc-lowering does look at the AllocaInst type ... sigh.

Reply:
I guess for now, we could just insert an appropriate amount of padding at the LLVM type if the allocation is not aligned?

Reply:
Or maybe we could stop rounding up the datatype size?

Reply:
I think we should stop rounding, since at best it may waste space that could be useful